### ISSUE TITLE
Proposed Analysis: fusion_filtering rerun for v7

### ISSUE THREAD

**Issue by kgaonkar6** [PediatricOpenTargets/ticket-tracker#126]
#### What analysis module should be updated and why?
We need to rerun fusion_filtering with the new v7 RNAseq files for data release.

#### What changes need to be made? Please provide enough detail for another participant to make the update.
Rerun

#### What input data should be used? Which data were used in the version being updated?
v7

**Comment by runjin326** [comment]
@kgaonkar6, just to double check, now when we run `Rscript 02-fusion-filtering.R`, `Rscript 03-Calc-zscore-annotate.R`, and `Rscript 06-recurrent-fusions-per-cancer-group.R` - cohort of interest will be 'PBTA,GMKF,TARGET' correct?

**Comment by kgaonkar6** [comment]
corrrect! I think we will need different gtex subsets per cancer type in TARGET?  cc @jharenza for additional info.

**Comment by kgaonkar6** [comment]
Just adding TARGET cancers
```
histology_df.loc[(histology_df.cohort=="TARGET") & (histology_df.experimental_strategy=="RNA-Seq"),'cancer_group'].value_counts()
Acute Myeloid Leukemia              839
Acute Lymphoblastic Leukemia        702
Neuroblastoma                       225
Wilms tumor                         133
Osteosarcoma                         87
Rhabdoid tumor                       73
NA                                   15
Clear cell sarcoma of the kidney     13

```

**Comment by runjin326** [comment]
Right - I also looked at the ones that we currently have RNA-seq data (by filtering on tumor_ploidy) 
`> histoloiges_df %>% filter(cohort == "TARGET") %>% filter(!is.na(tumor_ploidy)) %>% pull(broad_histology) %>% unique() %>% as.character()
[1] "Neuroblastoma"          "Acute Myeloid Leukemia"`

For this release, maybe we should separate those and for neuroblastoma still use `Adrenal Gland` and for AML use `Bone Marrow`? But after we process more, we might need to think about a more systematic ways to deal with that.

**Comment by jharenza** [comment]
We should have all TARGET RMA-Seq - ploidy comes from the DNA data and only NBL/AML DNA were run for v7.

I can gather the controls which should be relevant later today.

**Comment by runjin326** [comment]
Oh yes sorry I was getting confused between the two modules. Thanks, @jharenza!

**Comment by kgaonkar6** [comment]
Thanks @jharenza !

@runjin326 thinking about this further. Before we can rerun the code I think we need a PR for fusion_filtering to update the `03` script for zscoring. Currently we only filter tpm file by cohort because previously each cohort matched 1 gtex_group( `PBTA`  with `Brain`gtex  and `GMKF` with `Adrenal`gtex) https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/3a1dd99454486cd4ec3d453cd74a3e506dd5f0d9/analyses/fusion_filtering/03-Calc-zscore-annotate.R#L194-L205 

But TARGET has multiple cancer_groups in cohort which will need different gtex_group files. 

Should we create a file to match each gtex_group to a set of `cancer_groups` so we can then filter the tpm for by samples that exist in the list of cancer_groups and use the matched gtex.

**Comment by kgaonkar6** [comment]
For similar tasks before we have used json files as input but feel free to use any other solutions to this. 

gtex_cancer_group.json file:
```
{
    "Brain": [
        "High-grade glioma/astrocytoma",
        "Ependymoma"
    ],
    "Adrenal": [
        "Neuroblastoma"
    ],
    "Bone Marrow": [
        "Acute Myeloid Leukemia"
    ],
}
```
in R you could read json files with `jsonlite` which is in the docker already like

```
# Read in file
check <- jsonlite::read_json("gtex_cancer_group.json")
# gather cancer_group matched with gtex_group == Brain
unname(unlist(check["Brain"]))
[1] "High-grade glioma/astrocytoma" "Ependymoma"  
```

**Comment by runjin326** [comment]
@kgaonkar6, gotcha, thanks! This is really helpful! I will prepare a PR accordingly!

**Comment by runjin326** [comment]
For the cancer_group vs. normal matching, for now I will match neuroblastoma to adrenal gland, all cancer groups in PBTA to brain and for cancer groups in TARGET, I will wait for @jharenza. All cancer groups in PBTA are listed here: 
`> histoloiges_df %>% filter(cohort=="PBTA") %>% pull(cancer_group) %>% unique()`
 [1] Ependymoma                                
 [2] <NA>                                      
 [3] High-grade glioma/astrocytoma             
 [4] Low-grade glioma/astrocytoma              
 [5] Rosai-Dorfman disease                     
 [6] Sarcoma                                   
 [7] Adamantinomatous Craniopharyngioma        
 [8] Ewing sarcoma                             
 [9] Medulloblastoma                           
[10] Schwannoma                                
[11] CNS Embryonal tumor                       
[12] Diffuse midline glioma                    
[13] Atypical Teratoid Rhabdoid Tumor          
[14] Choroid plexus papilloma                  
[15] Embryonal tumor with multilayer rosettes  
[16] Neurofibroma/Plexiform                    
[17] Langerhans Cell histiocytosis             
[18] Choroid plexus carcinoma                  
[19] Primary CNS lymphoma                      
[20] Meningioma                                
[21] Teratoma                                  
[22] Dysembryoplastic neuroepithelial tumor    
[23] Hemangioblastoma                          
[24] Diffuse intrinsic pontine glioma          
[25] Adenoma                                   
[26] Metastatic secondary tumors;Neuroblastoma 
[27] Epilepsy                                  
[28] Metastatic secondary tumors               
[29] Subependymal Giant Cell Astrocytoma       
[30] Ganglioneuroblastoma                      
[31] Craniopharyngioma                         
[32] Dysplasia/Gliosis;Glial-neuronal tumor NOS
[33] Malignant peripheral nerve sheath tumor   
[34] Pineoblastoma                             
[35] Glial-neuronal tumor NOS                  
[36] Rhabdomyosarcoma                          
[37] Myxoid spindle cell tumor                 
[38] Chordoma                                  
[39] Germinoma                                 
[40] Ganglioneuroma                            
[41] Extraventricular Neurocytoma              
[42] CNS neuroblastoma                         
[43] Ganglioglioma                             
[44] Arteriovenous malformation                
[45] Melanocytic tumor                         
[46] Fibromyxoid lesion                        
[47] Oligodendroglioma                         
[48] Central Neurocytoma                       
[49] Choroid plexus cyst                       
[50] Cavernoma                                 
[51] Neuroblastoma                             
[52] Germinoma;Teratoma                        
[53] Reactive connective tissue                
[54] Myofibroblastoma

**Comment by jharenza** [comment]
sorry, I forgot about this last night!

See below - we do not have bone marrow in gtex, so I guess we can use blood. We don't have bone for osteo, so that one may have to have no control. 

```
Acute Myeloid Leukemia              Blood
Acute Lymphoblastic Leukemia        Blood
Neuroblastoma                       Adrenal Gland
Wilms tumor                         Kidney
Osteosarcoma                        NA (no bone in Gtex)
Rhabdoid tumor                      Kidney
Clear cell sarcoma of the kidney    Kidney
```

**Comment by runjin326** [comment]
@jharenza, thanks for the information! So for the ones that do not have control, how do we do z-score calculation? Do we use the matrix's own means and SDs to do the calculation?

**Comment by jharenza** [comment]
I would say leave them out of that zscore analysis

**Comment by runjin326** [comment]
Gotcha!

**Comment by runjin326** [comment]
I have opened a new ticket to deal with the normal matrix issue https://github.com/PediatricOpenTargets/ticket-tracker/issues/131

**Comment by jharenza** [comment]
closing with [PR 61](https://github.com/PediatricOpenTargets/OpenPedCan-analysis/pull/61)
